src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java

```diff
@@ -90,6 +91,34 @@ where TYPE(e) = QuizExercise
             """)
     List<Course> findAllWithQuizExercisesWithEagerExercises();
 
+    @Query("""
+            select c.studentGroupName
+            from Course c
+            where c.id = :courseId
+            """)
+    String findStudentGroupName(@Param("courseId") long courseId);
+
+    @Query("""
+            select s.submissionDate as day, u.login as username
```

Comment:
I am not saying you should do this definitely but imo make your code a lot cleaner by:

- Create a Custom Class with the attributes you want, lets call it ActiverUserDTO with a constructor that expects the day and the username. Overwrite .equals and hashcode to compare day and username
- make the query return the custom class by doing something like `SELECT DISTINCT new com.mypackage.ActiveUserDTO(s.submissionDate, u.login) from ...` **This should remove duplicates and you can remove the whole remove duplicates from map method**